File: src/mission_control_lib/mission_control_lib/dummy_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class DummyPublisher(Node):
    def __init__(self):
        super().__init__('dummy_publisher')
        
        # Create a publisher to DUMMY_TOPIC
        self.publisher_ = self.create_publisher(String, 'DUMMY_TOPIC', 10)
        self.timer = self.create_timer(1.0, self.publish_dummy)
        self.count = 0

    def publish_dummy(self):
        msg = String()
        self.count += 1
        msg.data = 'Hello from dummy publisher!' + str(self.count)
        self.publisher_.publish(msg)
        self.get_logger().info(f"Publishing: {msg.data}")

def main(args=None):
    rclpy.init(args=args)

    np_node = DummyPublisher()
    rclpy.spin(np_node)
    
    np_node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/mission_control_cpp/src/dummy_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class DummyPublisher : public rclcpp::Node
{
public:
    DummyPublisher() : Node("dummy_publisher"), count_(0)
    {
        publisher_ = this->create_publisher<std_msgs::msg::String>("DUMMY_TOPIC", 10);
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&DummyPublisher::publish_dummy, this)
        );
    }

private:
    void publish_dummy()
    {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello from dummy publisher! " + std::to_string(++count_);
        RCLCPP_INFO(this->get_logger(), "Publishing: %s", msg.data.c_str());
        publisher_->publish(msg);
    }

    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    int count_;
};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<DummyPublisher>());
    rclcpp::shutdown();
    return 0;
}
